src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -530,7 +533,7 @@ public void run() {
                 }
                 logger.warn("[{}] deleting dangling index", metaData.index());
                 try {
-                    indicesService.deleteIndexStore("deleting dangling index", metaData);
+                    indicesService.deleteIndexStore("deleting dangling index", metaData, clusterService.state());
```

Comment:
if we pass the state IMO we should document what state it is? I also wonder if we should get the metadata from the state direclty  instead of passing it. And then just pass the index name?